tests/test_bgp_af.rb

```diff
@@ -149,83 +199,191 @@ def test_default_information_originate
+    elsif platform == :nexus
+      pattern = /^ *client-to-client reflection$/
+      #
+      # Default is 'client-to-client' is configured
+      #
+      af_string = get_bgp_af_cfg(asn, vrf, af)
+
+      assert_match(pattern, af_string,
+                   "Error: 'client-to-client reflection' is not configured " \
                    'and should be')
 
-    assert(bgp_af.client_to_client,
-           "Error: 'client-to-client is not configured but should be")
-    #
-    # Unset and verify
-    #
-
-    # Do a 'no client-to-client reflection'
-    bgp_af.client_to_client = false
-    pattern = /^ *no client-to-client reflection$/
-    af_string = get_bgp_af_cfg(asn, vrf, af)
-
-    assert_match(pattern, af_string,
-                 "Error: 'no client-to-client' is not configured and should be")
+      assert(bgp_af.client_to_client,
+             "Error: 'client-to-client is not configured but should be")
+    end
 
     #
     # Set and verify
     #
-
-    # Do a 'client-to-client reflection'
     bgp_af.client_to_client = true
-    af_string = get_bgp_af_cfg(asn, vrf, af)
+    if platform == :ios_xr
+      af_string = @device.cmd("show run router bgp #{asn} address-family " \
+                              'ipv4 unicast bgp client-to-client ' \
+                              'reflection disable')
+
+      pattern = /^ *bgp client-to-client reflection disable$/
+      assert_match(pattern, af_string,
+                   'bgp client-to-client disable is configured and should be')
+    elsif platform == :nexus
+      # Do a 'client-to-client reflection'
+      af_string = get_bgp_af_cfg(asn, vrf, af)
+      pattern = /^ *client-to-client reflection$/
+
+      refute_match(pattern, af_string,
+                   "Error: 'no client-to-client' " \
+                   'is configured and should not be')
+    end
 
-    refute_match(pattern, af_string,
-                 "Error: 'no client-to-client' is configured and should not be")
+    #
+    # Unset and verify
+    #
+    bgp_af.client_to_client = false
+    if platform == :ios_xr
+      af_string = @device.cmd("show run router bgp #{asn} address-family " \
+                              'ipv4 unicast bgp client-to-client ' \
+                              'reflection disable')
+      pattern = /^ *bgp client-to-client reflection disable$/
+      refute_match(pattern, af_string,
+                   'bgp client-to-client disable is configured ' \
+                   'but should not be')
+    elsif platform == :nexus
+      # Do a 'no client-to-client reflection'
+      pattern = /^ *no client-to-client reflection$/
+      af_string = get_bgp_af_cfg(asn, vrf, af)
+
+      assert_match(pattern, af_string,
+                   "Error: 'no client-to-client' " \
+                   'is not configured and should be')
+    end
   end
 
   ##
-  ## next_hop route-map
+  ## Negative tests with non-default vrf on ios_xr
   ##
-  def test_next_hop_route_map
+  def next_hop_route_policy_vrf_xr_neg_test
```

Comment:
This is going to run on a nexus too and will fail.
I think these vrf cases need to be wrapped into the existing test code and have a platform check for XR.

if xr
  #test getter and setter for a non-default vrf
  assert/raise
  next
end